torchao/prototype/moe_training/utils.py

```diff
@@ -145,6 +145,41 @@ def _to_2d_jagged_float8_tensor_rowwise(
     return x_fp8, x_scales
 
 
+def torch_to_3d_rowwise_float8_transpose_rhs(
+    input_hp: torch.Tensor,  # (E, K, N)
+    target_dtype: torch.dtype = torch.float8_e4m3fn,
+    round_scales_to_power_of_2: bool = False,
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    """
+    This function converts the 3D input tensor to a float8 tensor, with scales computed along logical columns
+    on a per-expert basis.
+
+    Args:
+        x (torch.Tensor): The input tensor to be converted to a float8 tensor. Shape (E, K, N).
+
+    Returns:
+        A tuple containing the float8 tensor and the scales used for the conversion.
+        Output shape: (E, N, K)
+        Scales shape: (E, 1, K
+    """
+    input_hp_t = input_hp.transpose(-2, -1)  # (E, N, K)
+    scales = tensor_to_scale(
+        input_hp_t,
+        target_dtype,
+        scaling_granularity=ScalingGranularity.AXISWISE,
+        axiswise_dim=-2,
+        round_scales_to_power_of_2=round_scales_to_power_of_2,
+    )  # (E, 1, K)
+
+    # Apply scales to tensor and convert to float8.
+    tensor_scaled = input_hp_t.to(torch.float32) * scales
+    float8_tensor = to_fp8_saturated(tensor_scaled, target_dtype)
```

Comment:
this is confusing because it sounds like `Float8TrainingTensor`, maybe name it `float8_data`?